samplesheets/tests/test_views_api_taskflow.py

```diff
@@ -144,6 +235,644 @@ def test_post_created(self):
+        self.assertEqual(obj.description, 'bla')
+
+    def test_create_invalid_data(self):
+        """Test creating a request with invalid data"""
+        self.assertEqual(IrodsDataRequest.objects.count(), 0)
+        self._assert_alert_count(CREATE_ALERT, self.user, 0)
+        self._assert_alert_count(CREATE_ALERT, self.user_delegate, 0)
+        post_data = {'path': '/doesnt/exist', 'description': 'bla'}
+        url = reverse(
+            'samplesheets:api_irods_request_create',
+            kwargs={'project': self.project.sodar_uuid},
+        )
+        with self.login(self.user_contrib):
+            response = self.client.post(url, post_data)
+            self.assertEqual(response.status_code, 400)
+        self.assertEqual(IrodsDataRequest.objects.count(), 0)
+
+    def test_create_invalid_path_assay_collection(self):
+        """Test creating a request with assay path (should fail)"""
+        self.assertEqual(IrodsDataRequest.objects.count(), 0)
+        post_data = {'path': self.assay_path, 'description': 'bla'}
+        url = reverse(
+            'samplesheets:api_irods_request_create',
+            kwargs={'project': self.project.sodar_uuid},
+        )
+        with self.login(self.user_contrib):
+            response = self.client.post(url, post_data)
+            self.assertEqual(response.status_code, 400)
+        self.assertEqual(IrodsDataRequest.objects.count(), 0)
+
+    def test_create_multiple(self):
+        """Test creating multiple requests"""
+        path2 = os.path.join(self.assay_path, TEST_FILE_NAME + '_2')
+        path2_md5 = os.path.join(self.assay_path, TEST_FILE_NAME + '_2.md5')
+        self.irods.data_objects.create(path2)
+        self.irods.data_objects.create(path2_md5)
+
+        self.assertEqual(IrodsDataRequest.objects.count(), 0)
+        self._assert_alert_count(CREATE_ALERT, self.user, 0)
+        self._assert_alert_count(CREATE_ALERT, self.user_delegate, 0)
+        url = reverse(
+            'samplesheets:api_irods_request_create',
+            kwargs={'project': self.project.sodar_uuid},
+        )
+        with self.login(self.user_contrib):
+            response = self.client.post(url, self.post_data)
+            self.assertEqual(response.status_code, 200)
+            self.post_data['path'] = path2
+            response = self.client.post(url, self.post_data)
+            self.assertEqual(response.status_code, 200)
+
+        self.assertEqual(IrodsDataRequest.objects.count(), 2)
+        self._assert_alert_count(CREATE_ALERT, self.user, 1)
+        self._assert_alert_count(CREATE_ALERT, self.user_delegate, 1)
+
+
+class TestIrodsRequestUpdateAPIView(TestIrodsRequestAPIViewBase):
+    """Tests for IrodsRequestUpdateAPIView"""
+
+    def test_update(self):
+        """Test updating a request"""
+        update_data = {'path': self.path, 'description': 'Updated'}
+
+        with self.login(self.user_contrib):
+            self.client.post(
```

Comment:
Same comment as in the view class itself, I think we should use PUT and possibly PATCH in this update view instead of POST, to keep things consistent.